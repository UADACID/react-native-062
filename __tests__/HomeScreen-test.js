import React from 'react';
import HomeScreen from '../src/screens/home/home.screen';
import { shallow } from 'enzyme';
// import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });
describe('sample test', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<HomeScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('should show counter title', () => {
        const wrapper = shallow(<HomeScreen />);
        expect(
            wrapper.findWhere((node) => node.prop('testID') === 'counter-title'),
        ).toExist();
    });
})