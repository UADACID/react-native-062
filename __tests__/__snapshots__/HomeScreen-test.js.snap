// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sample test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    testID="counter-title"
  >
    counter request : 
    0
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        disabled={false}
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Request to API 500
      </Text>
    </View>
  </View>
</View>
`;

exports[`sample test should show counter title 1`] = `ShallowWrapper {}`;
